lib/perl/Genome/InstrumentData/AlignmentResult/Speedseq.pm

```diff
@@ -13,13 +13,82 @@ class Genome::InstrumentData::AlignmentResult::Speedseq {
     ],
 };
 
-sub _run_aligner {
+sub post_create {
+    return 1;
+}
+
+sub get_bam_file {
+    my $self = shift;
+
+    # Create doesn't do all of the necessary post-processing. This is being
+    # delayed until the first time that the bam file is revivified.
+    # If we don't have an allocation then this is the first revivification and
+    # we will need to do post-processing.
+    my $guard = $self->get_speedseq_bam_lock->unlock_guard();
+    unless ($self->disk_allocations) {
+        return $self->_inititalize_revivified_bam;
+    }
+    else {
+        return $self->SUPER::get_bam_file;
+    }
+}
+
+sub get_speedseq_bam_lock {
```

Comment:
The lock resource must be different from the one used by revivification.  Otherwise they will conflict with each other yielding a `Tried to lock resource more than once` error, since revivification is called as part of the Speedseq creation process.  If it would reduce confusion, they could be made less similar.